src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ViewExecutor.cs

```diff
@@ -130,11 +130,11 @@ public class ViewExecutor
                 throw new ArgumentNullException(nameof(view));
             }
 
-            var services = actionContext.HttpContext.RequestServices;
             if (viewData == null)
             {
+                var services = actionContext.HttpContext.RequestServices;
                 var metadataProvider = services.GetRequiredService<IModelMetadataProvider>();
-                viewData = new ViewDataDictionary(metadataProvider);
+                viewData = new ViewDataDictionary(metadataProvider, actionContext.ModelState);
```

Comment:
`ModelState` is a get-only property that can never be `null`. Therefore, no need to call the `internal` constructor.